consensus/fork.go

```diff
@@ -4,119 +4,115 @@ import (
 	"github.com/NebulousLabs/Sia/crypto"
 )
 
-// backtrackToBlockchain returns a list of nodes that go from the current node
-// to the first common parent. The common parent will be the final node in the
-// slice.
-func (s *State) backtrackToBlockchain(bn *blockNode) (nodes []*blockNode) {
-	nodes = append(nodes, bn)
+// invalidateNode recursively moves the children of a blockNode to the bad
+// blocks list.
+func (s *State) invalidateNode(node *blockNode) {
+	for i := range node.children {
+		s.invalidateNode(node.children[i])
+	}
+
+	delete(s.blockMap, node.block.ID())
+	s.badBlocks[node.block.ID()] = struct{}{}
+}
+
+// backtrackToBlockchain traces backwards from 'bn' until it reaches a node in
+// the State's current path (the "common parent"). It returns the (inclusive)
+// set of nodes between the common parent and 'bn', starting from the former.
+func (s *State) backtrackToBlockchain(bn *blockNode) []*blockNode {
+	path := []*blockNode{bn}
 	for s.currentPath[bn.height] != bn.block.ID() {
 		bn = bn.parent
-		nodes = append(nodes, bn)
+		// prepend, not append
+		path = append([]*blockNode{bn}, path...)
```

Comment:
hmm.

Do me a favor and closely double check everywhere (grep style) that calls backtrackToBlockchain to make sure that you adjusted everything since you reversed the order that the blocks get returned.